src/main/UserActivityMonitor.js

```diff
@@ -0,0 +1,170 @@
+// Copyright (c) 2015-2016 Yuya Ochiai
+// Copyright (c) 2016-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+'use strict';
+
+import EventEmitter from 'events';
+
+import electron from 'electron';
+
+const {app} = electron;
+
+/**
+ * Monitors system idle time, listens for system events and fires status updates as needed
+ */
+export default class UserActivityMonitor extends EventEmitter {
+  constructor() {
+    super();
+
+    this.isActive = true;
+    this.forceUserIsInactive = false;
+    this.idleTime = 0;
+    this.lastStatusUpdate = 0;
+    this.systemIdleTimeIntervalID = -1;
+
+    this.config = {
+      regularUpdateFrequencyMs: 1 * 1000,
+      statusUpdateThresholdMs: 5 * 1000,
+      userActivityTimeoutSec: 5 * 60,
+    };
+
+    this.handleSuspend = this.handleSuspend.bind(this);
```

Comment:
Do these need to be bound manually? I was under the assumption that we only did that in the apps because of how `React.Component` is written internally. Alternatively, we could define the methods as arrow functions since that makes manual binding unnecessary.

Reply:
Without the binding, errors are thrown and fat arrow functions aren't supported by the current setup.

Reply:
Added inline comment to that effect